src/default.nix

```diff
@@ -251,6 +251,7 @@ in let
     builder,
     builderArgs,
     fetchedSources,
+    sourceRoot,
```

Comment:
I think this can be removed. It isn't used by the builders. Also I think that we should change the way we handle relative path sources. I think we should always pass the root source as `src` and then set `sourceRoot` of `mkDerivation`. The unpackPhase will then switch to the correct subdir. That would allow packages to always refer to files in parent directories without issues. Some nodejs builds currently crash because sub-projects want to access files of parent projects in order to build.

Reply:
It is used by the Rust builders with this PR, to be able to get the full project repository so path dependencies work properly. Before it would only capture the path dependency directory, which will break if the path dependency depends on other path dependencies that are in a parent directory. We could remove this fine if we removed the `builtins.path` usage for path depedencies, but I think that would cause regressions?

Reply:
True. This is a trade-off situation that should be carefully thought through.
The isolation of sources gives us better incremental builds, but it also breaks some builds. We should probably provide the user with the choice to turn this on or off, and have the default on the safer side which is probably not to do source isolation. At least for nodejs that would be the case. Maybe the default can be different from subsystem to subsystem.

Reply:
Yes, I agree. I think we should have different functions for getting an isolated path vs a non-isolated path. And then the builder can decide which one to use.